tests/functional/local/apigw/test_service.py

```diff
@@ -450,6 +502,33 @@ def test_post_binary_with_incorrect_content_type(self):
         self.assertEquals(response.headers.get('Content-Type'), "application/json")
 
 
+class TestService_CaseInsensiveDict(TestCase):
+    def test_contains(self):
+        data = CaseInsensitiveDict({
+            'Content-Type': 'text/html',
+            'Browser': 'APIGW',
+        })
+
+        self.assertTrue('content-type' in data)
+        self.assertTrue('Content-Type' in data)
+        self.assertTrue('CONTENT-TYPE' in data)
+        self.assertTrue('Browser' in data)
+        self.assertTrue('Dog-Food' not in data)
```

Comment:
We try and keep unit tests to only one assert as much as possible. There are always exceptions but generally if you are testing two different cases, even if they are related, you should split them out into different tests.

Unit tests are cheep and fast. While this seems very much copy-paste, it leads to better isolated tests in overall. The main advantage here is when you make code changes that impact multiple asserts. When chaining asserts like this in one test, the test will fail when the first assert fails. This gives no indication of the bigger picture and state of your system. Devs following TDD constantly run unit test and rely heavily on this idea. I should be able to run the tests and see/understand everything that went wrong. 

You can very easily parameterize this and the following function to achieve this. Here is an example how we are doing this already: https://github.com/awslabs/aws-sam-cli/blob/develop/tests/unit/local/apigw/test_service.py#L520